pennylane_lightning/lightning_qubit/_measurements.py

```diff
@@ -401,31 +401,6 @@ def _process_single_shot(samples):
 
             return tuple(processed)
 
-        # if there is a shot vector, build a list containing results for each shot entry
-        if shots.has_partitioned_shots:
-            processed_samples = []
-            for s in shots:
-                # currently we call sample_state for each shot entry, but it may be
-                # better to call sample_state just once with total_shots, then use
-                # the shot_range keyword argument
-                try:
-                    if self._mcmc:
-                        samples = self._measurement_lightning.generate_mcmc_samples(
-                            len(wires), self._kernel_name, self._num_burnin, s
-                        ).astype(int, copy=False)
-                    else:
-                        samples = self._measurement_lightning.generate_samples(
```

Comment:
Surprised that it performs better even we don't use C++ APIs. Could you please clarify it?

Reply:
We now calculate samples for the sum(shots_vector) by calling the C++ function only once, instead of multiple calls for each shots in shots_vector. That's the main reason behind this 1e3+X performance improvement in the computation of `qml.sample` and `qml.counts`.

Reply:
:100: